python/hail/history.py

```diff
@@ -0,0 +1,170 @@
+from __future__ import print_function  # Python 2 and 3 print compatibility
+
+import datetime
+import inspect
+import itertools
+from collections import OrderedDict
+from decorator import decorator
+from hail.utils import hadoop_write, wrap_to_list
+from hail.java import Env
+
+
+def parse_args(f, args, kwargs):
+    argspec = inspect.getargspec(f)
+    kwargs_not_default = OrderedDict({})
+
+    if argspec.defaults:
```

Comment:
see the comment below about writing as kwargs all the time. That might mean we don't need this extra complexity